e2e/e2e_test.go

```diff
@@ -87,17 +108,202 @@ var _ = Describe("Whereabouts functionality", func() {
 				Expect(inRange(ipv4TestRange, secondaryIfaceIP)).To(Succeed())
 			})
 		})
+
+		Context("Replicaset tests", func() {
+			var (
+				replicaCount int32
+				dt           time.Time
+				k8sClient    *kubeClient.Client
+				label        string
+			)
+
+			BeforeEach(func() {
+				By("creating a replicaset with whereabouts net-attach-def")
+				replicaCount = int32(0) // begin test with 0 replicas.
+				dt = time.Now()
+				label = wbLabelEqual
+
+				k8sClient, err = kubeClient.NewClientViaKubeconfig(envVars.kubeconfigPath)
+				Expect(err).NotTo(HaveOccurred())
+
+				replicaSet, err = clientInfo.provisionReplicaSet(
+					replicaCount,
+					podTierLabel(rsName),
+					podNetworkSelectionElements(testNetworkName),
+				)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				By("deleting replicaset with whereabouts net-attach-def")
+				Expect(clientInfo.deleteReplicaSet(replicaSet)).To(Succeed())
+			})
+
+			It("allocates each IP pool entry with a unique pod IP", func() {
+				By("creating max number of pods and checking IP Pool validity")
+
+				for i := 0; i < envVars.numThrashIter; i++ {
+					dt = time.Now()
+					fmt.Printf("iteration %d of %d, %s\n", i, envVars.numThrashIter, dt.String())
+
+					// Set to 0 replicas
+					replicaCount = int32(0)
+					replicaSet, err = clientInfo.updateReplicaSet(replicaSetObject(
+						replicaCount,
+						podTierLabel(rsName),
+						podNetworkSelectionElements(testNetworkName),
+					))
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(WaitForReplicaSetSteadyState(clientInfo.Client, testNamespace, label, replicaSet, rsSteadyTimeout)).To(Succeed())
+
+					// Check for 0 IP pool allocations
+					ipPools, err := k8sClient.ListIPPools(context.Background())
+					Expect(err).To(BeNil())
+					Expect(ipPools[0].Allocations()).To(BeEmpty()) // get ipPool by name rather than by hoping index is right - TODO
+
+					// Increase to desired number of replicas for test
+					allPods, err := clientInfo.Client.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					maxReplicas := (envVars.numComputeNodes*maxPodsPerNode - (len(allPods.Items))) * envVars.fillPercentCapacity / 100
+					replicaCount = int32(maxReplicas)
```

Comment:
you could inline these 2. I don't think the `maxReplicas` is being used anywhere else - it's just an intermediate value.